tools/rostopic/src/rostopic/__init__.py

```diff
@@ -187,6 +187,10 @@ def _rostopic_hz(topic, window_size=-1, filter_expr=None):
     :param window_size: number of messages to average over, -1 for infinite, ``int``
     :param filter_expr: Python filter expression that is called with m, the message instance
     """
+
+    if rospy.get_param('use_sim_time', False):
+        print("WARNING: topic [%s] may be using simulated time\n" % (topic))
```

Comment:
You are right, it only gets printed once since the function loop internally.

Can you print the warning to `stderr` please.